compiler/src/dotty/tools/dotc/sbt/ExtractDependencies.scala

```diff
@@ -306,6 +306,13 @@ private class ExtractDependenciesCollector extends tpd.TreeTraverser { thisTreeT
       }
     }
 
+  private def addInheritanceDependencies(tree: Closure)(using Context): Unit =
+    // If the tpt is empty, this is a non-SAM lambda, so no need to register
+    // an inheritance relationship.
+    if !tree.tpt.isEmpty then
```

Comment:
should we still record a dependency to the specific `scala.Function<N>`?

Reply:
Technically yes, but it only matters if we break binary-compatibility of FunctionN, so I think it's worth skipping the dependency as an optimization.

Reply:
I agree :)